contents/tutorials/nuxt-surveys.md

```diff
@@ -0,0 +1,553 @@
+
+</script>
+```
+
+#### 3. Add the logic for displaying and hiding it.
+
+We want to make sure we don't show the survey again to users who have either submitted or dismissed it. We use [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to store this data and use it to check whether to show the survey or not.
+
+```vue file=app.vue
+<template>
+  <!-- ... rest of your template ... -->
+</template>
+
+<script setup>
+
+import { ref, onMounted } from 'vue';
+import { useNuxtApp } from '#app';
+
+const showSurvey = ref(false); // updated to false
+
+// rest of your code
+
+const checkSurveyInteraction = () => {
+  const hasInteractedWithSurvey = localStorage.getItem(`hasInteractedWithSurvey_${surveyID.value}`);
+  showSurvey.value = !hasInteractedWithSurvey;
+};
+
+onMounted(async () => {
+  await fetchActiveSurveys();
+  checkSurveyInteraction();
+});
+
+const fetchActiveSurveys = () => {
+  const { $posthog } = useNuxtApp();
+    $posthog().getActiveMatchingSurveys((surveys) => {
+    if (surveys.length > 0) {
+      const survey = surveys[0];
+      surveyID.value = survey.id;
+      surveyTitle.value = survey.questions[0].question;
+      checkSurveyInteraction();
+    }
+  });
+};
+
+const handleDismiss = () => {
+  showSurvey.value = false;
+  localStorage.setItem(`hasInteractedWithSurvey_${surveyID.value}`, 'true');
+};
+
+const handleSubmit = (value) => {
+  showSurvey.value = false;
+  localStorage.setItem(`hasInteractedWithSurvey_${surveyID.value}`, 'true');
+};
+
+</script>
+```
+
+#### 4. Capture interactions from it.
+
+The final step in setting up our survey is capturing interactions. This enables us to analyze the results in PostHog. 
+
+There are 3 events to capture:
+
+1. `"survey shown"`
+2. `"survey dismissed"`
+3. `"survey sent"` (for responses)
+
+You can capture these events using `this.$posthog.capture()`:
+
+```vue file=app.vue
+<template>
+  <!-- ... rest of your template ... -->
+</template>
+
+<script setup>
+
+// rest of your code
+
+const fetchActiveSurveys = () => {
+  const { $posthog } = useNuxtApp();
+    $posthog().getActiveMatchingSurveys((surveys) => {
+    if (surveys.length > 0) {
+      const survey = surveys[0];
+      surveyID.value = survey.id;
+      surveyTitle.value = survey.questions[0].question;
+      checkSurveyInteraction();
+      if (showSurvey) {
+        $posthog().capture("survey seen", {
+          $survey_id: surveyID // required
+        })
+      }
+    }
+  });
+};
+
+const handleDismiss = () => {
+  showSurvey.value = false;
+  localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+
+  const { $posthog } = useNuxtApp();
+  $posthog().capture("survey dismissed", {
+    $survey_id: surveyID // required
+  });
+};
+
+const handleSubmit = (value) => {
+  showSurvey.value = false;
+  localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+
+  const { $posthog } = useNuxtApp();
+  $posthog().capture("survey sent", {
```

Comment:
Just checked this event and it is showing this error:
![CleanShot 2024-01-16 at 20 16 37@2x](https://github.com/PostHog/posthog.com/assets/34755028/1a1516a8-94d4-4de7-8e5e-75eec5db238a)

Also not showing as a response in results, might be something to check to make sure it works properly.

Reply:
Interestingly this is a `number` on my PostHog instance. Maybe it was string and then later changed to number? @neilkakkar any idea what the correct type is supposed to be? 

<img width="750" alt="Screenshot 2024-01-17 at 8 10 10 PM" src="https://github.com/PostHog/posthog.com/assets/7090054/29bd0fd6-aa9e-4122-a8c7-1a9d70969e34">

Reply:
From the [slack conversation](https://posthog.slack.com/archives/C034XD440RK/p1705568310771809), this should be updated to number. I'll do this in a different PR, as there are a few other tutorials this will need to be updated for too (and also the docs)